### PR TITLE
refined migrating mariadb procedure

### PR DESCRIPTION
**Pull request by klgill**
https://issues.redhat.com/browse/RHOSPDOC-1645

@bogdando or @jistr Please address the questions I have added to this PR. Thank you!

### CODE REVIEW

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc [hunk 1]
```diff
@@ -1,27 +1,22 @@
...
+//[NOTE]
+//This example scenario describes a simple single-cell setup. Real
+//multi-stack topology recommended for production use results in different
+//cells DBs layout, and should be using different naming schemes (not covered
+//here this time). kgilliga: I hid the same note in the Compute adoption procedure. Will likely reinstate this after multi-cell is released.
 
 .Prerequisites
 
-* Make sure the previous Adoption steps have been performed successfully.
- ** The `OpenStackControlPlane` resource must be already created.
```
Comment (klgill) [inline docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:19]: Is “* The `OpenStackControlPlane` resource must be already created,” a prerequisite for the entire adoption guide, not just this procedure?

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc [hunk 2]
```diff
@@ -52,13 +47,13 @@ SOURCE_GALERA_MEMBERS=(
 SOURCE_DB_ROOT_PASSWORD=$(cat ~/tripleo-standalone-passwords.yaml | grep ' MysqlRootPassword:' | awk -F ': ' '{ print $2; }')
 ----
 
-* Prepare MariaDB copy directory and the adoption helper pod
+* Prepare the MariaDB copy directory and the adoption helper pod:
 
-* Create a temporary folder to store adoption helper pod (pick storage requests to fit MySQL database size):
+. Create a temporary folder to store the adoption helper pod. Choose storage requests that fit the MySQL database size:
```
Comment (klgill) [inline docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:52]: Does the customer need to do something? E.g, enter a value in the “requests:” parameter?

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc [hunk 3]
```diff
@@ -131,38 +126,32 @@ done
...
 +
 [NOTE]
-You need to transition {compute_service_first_ref} services imported later on into a
-superconductor architecture. For that, delete the old service records in
-cells DBs, starting from the cell1. New records will be registered with
-different hostnames provided by the {compute_service} service operator. All {compute_service}
-services, except the compute agent, have no internal state, and its service
-records can be safely deleted. You also need to rename the former `default` cell
-to `cell1`.
+You must transition {compute_service_first_ref} services imported later on into a superconductor architecture by deleting the old service records in cells databases, starting with `cell1`. New records are registered with different hostnames provided by the {compute_service} operator. All Compute services, except the Compute agent, have no internal state, and its service records can be safely deleted. You also need to rename the former `default` cell to `cell1`.
```
Comment (klgill) [inline docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:149]: “Later on” means when? During the Compute service adoption procedure?

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc [hunk 4]
```diff
@@ -226,10 +215,10 @@ EOF
 
 .Verification
 
-Compare the following outputs with the topology specific configuration.
+Compare the following outputs with the topology-specific service configuration.
```
Comment (klgill) [inline docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:218]: In the “Verification” section, it says “Compare the following outputs with the topology-specific service configuration.” “Does “the following outputs” refer to the outputs from the commands in the following steps, i.e., checking that the database imported correctly, etc.?

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc [hunk 5]
```diff
@@ -267,14 +256,16 @@ novadb_svc_records=$(oc exec openstack-cell1-galera-0 -c galera -- mysql -rs -ur
...
 
-. During the pre/post checks the pod `mariadb-client` might have returned a pod security warning
-related to the `restricted:latest` security context constraint. This is due to default security
-context constraints and will not prevent pod creation by the admission controller. You'll see a
-warning for the short-lived pod but it will not interfere with functionality.
-. Delete the `mariadb-data` pod and `mariadb-copy-data` persistent volume claim with databases backup (consider making a snapshot of it, before deleting)
+. Delete the `mariadb-data` pod and `mariadb-copy-data` persistent volume claim with databases backup:
++
+[NOTE]
+Consider taking a snapshot of them before deleting.
```
Comment (klgill) [inline docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:262]: “Consider taking a snapshot before deleting” - by using the snapshot and revert tool? https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/17.1/html/backing_up_and_restoring_the_undercloud_and_control_plane_nodes/assembly_snapshot-and-revert-appendix_snapshot-and-revert-appendix

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc (hunk 1 above)
Comment (jistr) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:19]: Except the initial step where the `OpenStackControlPlane` is created, yes.

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc [hunk 6]
```diff
@@ -131,38 +126,32 @@ done
 . Get the count of not-OK source databases:
 +
 ----
-oc rsh mariadb-copy-data mysql -h "${SOURCE_MARIADB_IP}" -uroot -p"${SOURCE_DB_ROOT_PASSWORD}" -e "SHOW databases;"
+$ oc rsh mariadb-copy-data mysql -h "${SOURCE_MARIADB_IP}" -uroot -p"${SOURCE_DB_ROOT_PASSWORD}" -e "SHOW databases;"
 ----
 
-. Run mysqlcheck on the original DB to look for things that are not OK:
+. Run `mysqlcheck` on the original database to check for things that are not OK:
```
[the remark this answers is not in the document]
Comment (jistr) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:132]: I'm not sure (never saw it happen myself). We'll probably want to tweak the snippets below so that error/success conditions are more obvious. There is a bug reported for that: https://issues.redhat.com/browse/OSPRH-7859

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc (hunk 3 above)
Comment (jistr) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:149]: cc @bogdando

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc (hunk 4 above)
Comment (jistr) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:218]: cc @bogdando (I'm not sure how this was meant.)

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc (hunk 5 above)
Comment (jistr) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:262]: @bogdando please correct me if needed, but i think it would be via something like: https://docs.redhat.com/en/documentation/openshift_container_platform/4.16/html-single/storage/index#lvms-about-volume-snapsot_logical-volume-manager-storage

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc [hunk 7]
```diff
@@ -267,14 +256,16 @@ novadb_svc_records=$(oc exec openstack-cell1-galera-0 -c galera -- mysql -rs -ur
...
-oc delete pod mariadb-copy-data
-oc delete pvc mariadb-data
+$ oc delete pod mariadb-copy-data
+$ oc delete pvc mariadb-data
 ----
-For more information, see https://learn.redhat.com/t5/DO280-Red-Hat-OpenShift/About-pod-security-standards-and-warnings/m-p/32502[About pod security standards and warnings].
+
+[NOTE]
+During the pre-checks and post-checks, the `mariadb-client` pod might return a pod security warning related to the `restricted:latest` security context constraint. This warning is due to default security context constraints and does not prevent the admission controller from creating a pod. You see a warning for the short-lived pod, but it does not interfere with functionality.
+For more information, see link:https://learn.redhat.com/t5/DO280-Red-Hat-OpenShift/About-pod-security-standards-and-warnings/m-p/32502[About pod security standards and warnings].
```
[the remark this answers is not in the document]
Comment (jistr) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:271]: I think it's fine to leave it upstream too.

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc (hunk 2 above)
Comment (jistr) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:52]: Sorry i skipped this earlier. The folder will be removed (it is unused), there is a bug for it, i'll post a patch.

The sentence about storage requests refers to this part of the snippet below:

```
  resources:
    requests:
      storage: 10Gi
```

The storage request needs to be big enough to fit the overcloud databases stored in MariaDB. In the example we request 10 gigabytes.

**Comment by klgill** [comment]
Note to self. I need to make sure the changes from the following PR are implemented in this PR: https://github.com/openstack-k8s-operators/data-plane-adoption/pull/541

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc (hunk 6 above)
[the remark this answers is not in the document]
Comment (klgill) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc]: @bogdando Could we reword it to: "Run `mysqlcheck` on the original database to check for errors"?

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc (hunk 3 above)
Comment (bogdando) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:149]: correct

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc (hunk 4 above)
Comment (bogdando) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:218]: yes

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc (hunk 5 above)
Comment (bogdando) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:262]: right, any backend for CSI snapshotter should work

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc (hunk 1 above)
Comment (bogdando) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:19]: I think we should create osp secrets before?

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc [hunk 8]
```diff
@@ -52,13 +47,13 @@ SOURCE_GALERA_MEMBERS=(
 SOURCE_DB_ROOT_PASSWORD=$(cat ~/tripleo-standalone-passwords.yaml | grep ' MysqlRootPassword:' | awk -F ': ' '{ print $2; }')
 ----
 
-* Prepare MariaDB copy directory and the adoption helper pod
+* Prepare the MariaDB copy directory and the adoption helper pod:
```
Comment (Jenny-Anne) [inline docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:50]: Is the colon needed here at the end of line 50?

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc [hunk 9]
```diff
@@ -131,38 +126,32 @@ done
...
 +
 [NOTE]
-You need to transition {compute_service_first_ref} services imported later on into a
-superconductor architecture. For that, delete the old service records in
-cells DBs, starting from the cell1. New records will be registered with
-different hostnames provided by the {compute_service} service operator. All {compute_service}
-services, except the compute agent, have no internal state, and its service
-records can be safely deleted. You also need to rename the former `default` cell
-to `cell1`.
+You must transition {compute_service_first_ref} services imported later on into a superconductor architecture by deleting the old service records in cells databases, starting with `cell1`. New records are registered with different hostnames provided by the {compute_service} operator. All Compute services, except the Compute agent, have no internal state, and its service records can be safely deleted. You also need to rename the former `default` cell to `cell1`.
```
Comment (Jenny-Anne) [inline docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:149]: You must transition {compute_service_first_ref} services imported later on... > You must transition {compute_service_first_ref} services that you import later...

or

You must transition {compute_service_first_ref} services that are imported later...

Comment (Jenny-Anne) [inline docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:149]: "in cells databases" > "in the databases for cells" or "in cells' databases"?

Comment (Jenny-Anne) [inline docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:149]: Does "its" refer to the Compute agent? If it does, maybe update "and its service records" to "and the service records of the Compute agent"?

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc (hunk 4 above)
Comment (Jenny-Anne) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:218]: I was going to ask the same question :) Possibly rephrase  to something like "Compare the command outputs from the following steps with the topology-specific service configuration"?

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc [hunk 10]
```diff
@@ -267,14 +256,16 @@ novadb_svc_records=$(oc exec openstack-cell1-galera-0 -c galera -- mysql -rs -ur
 diff -Z <(echo $novadb_svc_records) <(echo $PULL_OPENSTACK_CONFIGURATION_NOVA_COMPUTE_HOSTNAMES)
 ----
 
-. During the pre/post checks the pod `mariadb-client` might have returned a pod security warning
-related to the `restricted:latest` security context constraint. This is due to default security
-context constraints and will not prevent pod creation by the admission controller. You'll see a
-warning for the short-lived pod but it will not interfere with functionality.
-. Delete the `mariadb-data` pod and `mariadb-copy-data` persistent volume claim with databases backup (consider making a snapshot of it, before deleting)
+. Delete the `mariadb-data` pod and `mariadb-copy-data` persistent volume claim with databases backup:
```
Comment (Jenny-Anne) [inline docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:259]: Is "databases backup" a tool or a command or something that deletes the pod and PVC? If possible, maybe specify what "databases backup" is to clarify the connection between it and the deletions.

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc [hunk 11]
```diff
@@ -131,38 +126,32 @@ done
 . Get the count of not-OK source databases:
```
Comment (Jenny-Anne) [inline docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:126]: If "not-OK" is not code, possibly rephrase to "Get the count of source databases that do not have an OK status:"

Comment (Jenny-Anne) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:126]: Just see `NOK` in the code snippets further down now so maybe revise my last comment to something like "Get the count of source databases with the `NOK` (not-OK), status:"

if that makes sense

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc (hunk 8 above)
Comment (klgill) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:50]: Yes, it's introducing some steps.

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc (hunk 9 above)
Comment (klgill) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:149]: Actually no. I think "its" refers to "all Compute services", so "its" should be changed to "their".

CC: @bogdando

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc (hunk 10 above)
Comment (klgill) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:259]: I don't think "databases backup" is a tool. 

@bogdando Is this step saying that users should consider backing up their databases, then delete the mariadb-data pod and mariadb-copy-data? We need to clarify the connection between the databases backup and the deletions.

Comment (bogdando) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:259]: the volume claim containing the database backup

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc (hunk 11 above)
Comment (klgill) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:126]: @bogdando Can you confirm that I can rewrite this step to say the following?
"Get the count of source databases with the NOK (not-OK) status:"

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc (hunk 9 above)
Comment (bogdando) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:149]: yes, that is correct

File: docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc (hunk 1 above)
Comment (klgill) [reply docs_user/modules/proc_migrating-databases-to-mariadb-instances.adoc:19]: I think that this is part of a broader conversation of what prerequisites are required before the entire adoption procedure is started, vs, what prerequisites are required for this specific procedure.

I don't think we need to delay merging this PR for that conversation to be resolved. We can revisit this procedure later if needed.
